examples/basic_query_extractor/src/main.rs

```diff
@@ -0,0 +1,141 @@
+//! A basic example application for working with the Gotham Query String Extractor
+
+extern crate futures;
+extern crate gotham;
+extern crate hyper;
+extern crate mime;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+extern crate serde_json;
+#[macro_use]
+extern crate gotham_derive;
+
+use hyper::{Response, StatusCode};
+use hyper::header::ContentType;
+
+use gotham::router::Router;
+use gotham::router::builder::{build_simple_router, DefineSingleRoute, DrawRoutes};
+use gotham::state::{State, FromState};
+
+/// `QueryParam` struct
+///
+/// It contains only a `name` field for the sake of simplicity
+/// It permits to use it as query parameter like:
+/// GET /widgets?name=t-shirt
+#[derive(Deserialize, StateData, StaticResponseExtender)]
+struct QueryParam {
+    name: String,
+}
+
+/// `Product` struct
+///
+/// It represents products that can be queried
+#[derive(Serialize)]
+struct Product {
+    name: String,
+    description: String,
+}
+
+// Generate a `Vec<Product>`
+fn generate_products() -> Vec<Product> {
+    vec![
+        Product {
+            name: "t-shirt".to_string(),
+            description: "t-shirt".to_string(),
+        },
+    ]
+}
+
+/// Returns a `Response` with the `Product` serialized to JSON
+/// If no `Product` is found, a `Response` containing a `StatusCode::NotFound`
+/// is returned.
+fn product_matcher(requested: &str, products: &Vec<Product>) -> Response {
+    match products.iter().find(|p| p.name == requested) {
+        Some(product) => {
+            Response::new()
+                .with_status(StatusCode::Ok)
+                .with_body(serde_json::to_string(product).unwrap())
+                .with_header(ContentType(mime::APPLICATION_JSON))
+        }
+        None => Response::new().with_status(StatusCode::NotFound),
+    }
```

Comment:
In both of these cases I recommend using the helper [create_response](https://github.com/gotham-rs/gotham/blob/388db28838a76b02b0c71fa71024d12bf6a8edd9/gotham/src/http/response/mod.rs#L20) instead of directly creating a `Response` object. The helper gives you a lot for free.